plugins/cassandra/cassandrawu.cpp

```diff
@@ -2470,6 +2304,169 @@ class CCassandraWorkUnit : public CPersistedWorkUnit
         }
     }
 
+    virtual void clearGraphProgress() const
+    {
+        const char *wuid = queryWuid();
+        CassandraBatch batch(cass_batch_new(CASS_BATCH_TYPE_UNLOGGED));
+        sessionCache->deleteChildByWuid(wuGraphProgressMappings, wuid, batch);
+        sessionCache->deleteChildByWuid(wuGraphStateMappings, wuid, batch);
+        sessionCache->deleteChildByWuid(wuGraphRunningMappings, wuid, batch);
+        CassandraFuture futureBatch(cass_session_execute_batch(sessionCache->querySession(), batch));
+        futureBatch.wait("clearGraphProgress");
+    }
+    virtual bool getRunningGraph(IStringVal &graphName, WUGraphIDType &subId) const
+    {
+        CassandraStatement statement(sessionCache->prepareStatement("SELECT graphID, subgraphID FROM wuGraphRunning where partition=? and wuid=?;"));
+        const char *wuid = queryWuid();
+        statement.bindInt32(0, rtlHash32VStr(wuid, 0) % NUM_PARTITIONS);
+        statement.bindString(1, wuid);
+        CassandraFuture future(cass_session_execute(sessionCache->querySession(), statement));
+        future.wait("select state");
+        CassandraResult result(cass_future_get_result(future));
+        if (cass_result_row_count(result))
+        {
+            const CassRow *row = cass_result_first_row(result);
+            assertex(row);
+            StringBuffer b;
+            getCassString(b, cass_row_get_column(row, 0));
+            graphName.set(b);
+            subId = getUnsignedResult(NULL, cass_row_get_column(row, 1));
+            return true;
+        }
+        else
+            return false;
+    }
+    virtual IConstWUGraphProgress *getGraphProgress(const char *graphName) const
+    {
+        CassandraStatement statement(sessionCache->prepareStatement("SELECT subgraphID, creator, progress FROM wuGraphProgress where partition=? and wuid=? and graphID=?;"));
+        const char *wuid = queryWuid();
+        statement.bindInt32(0, rtlHash32VStr(wuid, 0) % NUM_PARTITIONS);
+        statement.bindString(1, wuid);
+        statement.bindString(2, graphName);
+        CassandraFuture future(cass_session_execute(sessionCache->querySession(), statement));
+        future.wait("select state");
```

Comment:
should probably say getGraphProgress. It's just used for tracing in case of error, so not really critical.